rich/syntax.py

```diff
@@ -484,8 +520,28 @@ def tokens_to_spans() -> Iterable[Tuple[str, Optional[Style]]]:
                 )
             if self.background_color is not None:
                 text.stylize(f"on {self.background_color}")
+
+        if self._stylized_ranges:
+            self._apply_stylized_ranges(code=code, text=text)
+
         return text
 
+    def stylize_range(
+        self, style: Style, start: SyntaxPosition, end: SyntaxPosition
+    ) -> None:
+        """
+        Adds a custom style on a part of the code, that will be applied to the syntax display when it's rendered.
+        Line numbers are 1-based, while column indexes are 0-based.
+
+        Args:
+            style (Style): The style to apply.
+            start (Tuple[int, int]): The start of the range, in the form `[line number, column index]`.
+            end (Tuple[int, int]): The end of the range, in the form `[line number, column index]`.
+        """
+        self._stylized_ranges.append(
+            SyntaxHighlightRange(style=style, start=start, end=end)
```

Comment:
I think it is better to use positional args here. Granted keyword args are better for self-documenting code, but consistent ordering helps.